src/Functions/toJSONString.cpp

```diff
@@ -35,14 +38,18 @@ namespace
             WriteBufferFromVector<ColumnString::Chars> json(data_to);
             for (size_t i = 0; i < input_rows_count; ++i)
             {
-                serializer->serializeTextJSON(*arguments[0].column, i, json, FormatSettings());
+                serializer->serializeTextJSON(*arguments[0].column, i, json, format_settings);
                 writeChar(0, json);
                 offsets_to[i] = json.count();
             }
 
             json.finalize();
             return res;
         }
+
+    private:
+        /// Affects only subset of part of settings related to json.
+        FormatSettings format_settings;
```

Comment:
Why we need to store this in class member ? Does it get modified somewhere ?

Reply:
I moved it to class member, because I think it's to heavy to initialize it on every block. It's not modified, added `const`.

Reply:
@CurtizJ yes, agree.